### ISSUE TITLE
Extend Mesh

### ISSUE THREAD

**Issue by pmueller2** [nutofem/nuto#227]
If you import a mesh from Gmsh you get a MeshGmsh object that saves beside a mesh also some Groups of ElementCollections (corresponding to Gmsh physical groups). I really like that feature and I think this is not related to Gmsh but should belong to our or an extended mesh class.

What do you think?

**Comment by TTitscher** [comment]
I like the idea of the extended mesh. But I am not sure, if that should be another class or not. I always like the separation. On the one hand, the user can choose his includes. E.g. UnitMesh? `#include <.../UnitMeshFem.h>` Gmsh? `#include <.../MeshGmsh.h>` And maybe now: `#include <.../MeshFemCompanion.h>`? :smile:
On the user hand, the user has to know that something like that exists beside the mesh.

Features could include:

- keep track of groups
- efficient node selection (based on spatial data structures)
- additional infos (that are commonly requested) like 
    - `meshCompanion.DofValueAt(Eigen::Vector(42, 12), displacement);`
    - `meshCompanion.DofValueAt(node, displacement);`
- (Maybe even perform `NodeMerge/Extract`?)

**Comment by vhirtham** [comment]
> Maybe even perform NodeMerge/Extract

Yes please. A function that spits out a Eigen::MatrixXd containing the correctly ordered dof values would be really nice and saves the "user" 2 additional conversion steps: 

- Merger.Extract -> ToEigen
- FromEigen -> Merger.Merge
